src/application.rs

```diff
@@ -161,8 +163,77 @@ impl CoLink {
         Ok(response.get_ref().jwt.clone())
     }
 
+    pub async fn _create_entry_bulk(
+        &self,
+        key_name: &str,
+        payload: &[u8],
+    ) -> Result<String, Error> {
+        let mut client = self._grpc_connect(&self.core_addr).await?;
+        let metadata_key = format!("{}::metadata", key_name);
+        let metadata_key_copy = metadata_key.clone();
+        let result = metadata_key.clone();
+        let num_chunks = payload.len() / CHUNK_SIZE + 1;
+
+        let create_request = generate_request(
+            &self.jwt,
+            StorageEntry {
+                key_name: metadata_key,
+                payload: (num_chunks.to_string() + "-locked").into_bytes(), // simply store the number of chunks in the metadata
+                ..Default::default()
+            },
+        );
+
+        client.create_entry(create_request).await?;
+
+        let mut offset = 0;
+        let mut chunk_id = 0;
+        while offset < payload.len() {
+            let chunk_size = if offset + CHUNK_SIZE as usize > payload.len() {
+                payload.len() - offset
+            } else {
+                CHUNK_SIZE as usize
+            };
+            let chunk = payload[offset..offset + chunk_size].to_vec();
+            let request = generate_request(
+                &self.jwt,
+                StorageEntry {
+                    key_name: format!("{}::{}", key_name, chunk_id),
+                    payload: chunk,
+                    ..Default::default()
+                },
+            );
+
+            client.create_entry(request).await?;
+
+            offset += chunk_size;
+            chunk_id += 1;
+        }
+
+        let update_request = generate_request(
+            &self.jwt,
+            StorageEntry {
+                key_name: metadata_key_copy,
+                payload: (num_chunks.to_string() + "-unlocked").into_bytes(),
+                ..Default::default()
+            },
+        );
+
+        client.update_entry(update_request).await?;
+
+        Ok(result)
+    }
+
     pub async fn create_entry(&self, key_name: &str, payload: &[u8]) -> Result<String, Error> {
         let mut client = self._grpc_connect(&self.core_addr).await?;
+        if key_name.contains("$") {
```

Comment:
Could you make a conditional compilation for this check? (so that it goes to a non-empty "magic mount" only when the feature is enabled)

Reply:
Done.